src/iter/italgos.c

```diff
@@ -556,3 +559,127 @@ double power(unsigned int maxiter,
 	return s;
 }
 
+
+
+
+/**
+ * First Order Primal Dual algorithm. Solves min_x F(Ax) + G(x)
+ *
+ * @param maxiter maximum number of iterations
+ * @param epsilon stop criterion
+ * @param tau primal step size
+ * @param sigma dual step size
+ * @param decay decay rate
+ * @param theta convex combination rate
+ * @param N size of input, x
+ * @param M size of transformed input, Ax
+ * @param vops vector ops definition
+ * @param op_forw forward operator, A
+ * @param op_adj adjoint operator, AH
+ * @param thresh1 primal prox function, e.g. prox_l2ball
+ * @param thresh2 dual prox function, e.g. prox_wavelet_thresh
+ * @param x initial estimate
+ * @param monitor callback function
+ */
+void pridu(unsigned int maxiter, float epsilon, float tau, float sigma, float theta, float decay,
```

Comment:
This is probably more a function of me not knowing my iteration algorithms well, but I would personally really like it if the documentation above the function definition mentioned what parameter F and G are (mentioned in the first line as F(Ax) + G(x)). I'm assuming it is related to thresh1 and thresh2.

Reply:
I could the comment to, "thresh1 primal proximal function for F", "thresh2 primal proximal function for G". Would that help?

Reply:
Yeah!